src/app/kbv/controllers/single-amount-question.test.js

```diff
@@ -0,0 +1,221 @@
+const BaseController = require("hmpo-form-wizard").Controller;
+const Controller = require("./single-amount-question");
+
+const presenters = require("../../../presenters");
+jest.mock("../../../presenters");
+const moment = require("moment");
+
+const {
+  API: {
+    PATHS: { QUESTION, ANSWER },
+  },
+} = require("../../../lib/config");
+
+describe("single-amount-question controller", () => {
+  let controller;
+  let req;
+  let next;
+  let res;
+
+  beforeEach(() => {
+    controller = new Controller({ route: "/test" });
+    req = global.req;
+    req.form.options = {
+      fields: {},
+    };
+    req.session.question = {
+      questionKey: "Q1",
+      hint: "l",
+      text: "t",
+    };
+    next = jest.fn();
+    res = jest.fn();
+  });
+
+  it("should be an instance of BaseController", () => {
+    controller = new Controller({ route: "/test" });
+    expect(controller).toBeInstanceOf(BaseController);
+  });
+
+  describe("#locals", () => {
+    beforeEach(() => {
+      req.session = {
+        question: {},
+      };
+      req.translate = jest.fn();
+    });
+
+    it("should set threeMonthsAgo in locals", (done) => {
+      const callback = jest.fn((err, locals) => {
+        expect(err).toBeNull();
+        expect(locals.threeMonthsAgo).toEqual(
+          moment().subtract(3, "months").format("DD MMMM YYYY")
+        );
+        done();
+      });
+
+      controller.locals(req, res, callback);
+    });
+
+    it("should call super.locals with req and res", () => {
+      const superLocals = jest.spyOn(BaseController.prototype, "locals");
+
+      controller.locals(req, res, next);
+
+      expect(superLocals).toHaveBeenCalledWith(req, res, expect.any(Function));
+    });
+
+    it("should set question.label and question.hint in locals", (done) => {
+      presenters.questionToLabel.mockReturnValue("Question Label");
+      presenters.questionToHint.mockReturnValue("Question Hint");
+
+      const callback = jest.fn((err, locals) => {
+        expect(err).toBeNull();
+        expect(locals.question.label).toBe("Question Label");
+        expect(locals.question.hint).toBe("Question Hint");
+        done();
+      });
+
+      controller.locals(req, res, callback);
+    });
+
+    it("should call callback with error if super.locals returns an error", (done) => {
+      const mockError = new Error("Some error");
+
+      const callback = jest.fn((err, locals) => {
+        expect(err).toEqual(mockError);
+        expect(locals).toBeUndefined();
+        done();
+      });
+
+      const superLocals = jest.spyOn(BaseController.prototype, "locals");
+      superLocals.mockImplementation((req, res, callback) => {
+        callback(mockError);
+      });
+
+      controller.locals(req, res, callback);
+    });
+  });
+
+  describe("#saveValues", () => {
+    beforeEach(() => {
+      req.session.tokenId = "session-id";
+      req.axios.post = jest.fn();
+    });
+
+    it("should call single-amount-question endpoint", async () => {
```

Comment:
I don't think we need this one now as it's covered by `should call question endpoint to get next question and store it in session` ?